omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,179 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/visitconfiguration")
+public class VisitConfigurationController2_0 extends BaseRestController {
+
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	@ResponseBody
+	public Object getCurrentConfiguration() {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		VisitConfiguration visitConfiguration = new VisitConfiguration();
+		visitConfiguration.setEnableVisits(getEnableVisitsValue(administrationService));
+		visitConfiguration
+				.setEncounterVisitsAssignmentHandler(administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER));
+		visitConfiguration.setStartAutoCloseVisitsTask(getAutoCloseVisitsTaskStartedValue(schedulerService));
+		visitConfiguration.setVisitTypesToAutoClose(getVisitTypesToAutoCloseValue(administrationService, visitService));
+
+		return ConversionUtil.convertToRepresentation(visitConfiguration, Representation.FULL);
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.OK)
+	public void updateCurrentConfiguration(@RequestBody VisitConfiguration newConfiguration) throws SchedulerException {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		EncounterService encounterService = Context.getEncounterService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		// validate
+		if (newConfiguration.getEnableVisits() && StringUtils.isEmpty(newConfiguration.getEncounterVisitsAssignmentHandler())) {
+			throw new IllegalRequestException("Encounter Visit assignment handler cannot be empty");
+		}
+
+		administrationService
+				.setGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.toString(newConfiguration.getEnableVisits()));
+
+		if (newConfiguration.getEnableVisits()) {
+			String newEncounterVisitsAssignmentHandler = newConfiguration.getEncounterVisitsAssignmentHandler();
+			if (isEncounterVisitsAssignmentHandlerValid(newEncounterVisitsAssignmentHandler, encounterService)) {
+				administrationService
+						.setGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER, newEncounterVisitsAssignmentHandler);
+			} else {
+				throw new IllegalPropertyException("Provided encounterVisitsAssignmentHandler class does not exist.");
```

Comment:
Can you include the invalid class in the exception message?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/88e2282688cd94863e7d66608d42a2597e3c9cad